### DISCUSSION TITLE
why is my composite command resulting in a strange distorted mask effect?

### DISCUSSION THREAD

**Discussion by soyebravat** [ImageMagick/ImageMagick#5174] [Digital Image Processing]
I'm trying to batch overlay backgrounds to designs with specific parameters but the results are coming out very strange with a gray mask type effect. 

The code I'm using is

`FOR %G IN ("*.png") DO magick "%G" background.gif ( *.png -resize 800x990 ) -gravity Center -geometry +30-160 -composite "output.%G"
`

Can somebody please help?

![output orange](https://user-images.githubusercontent.com/33333343/170553977-8d79d797-30f8-46cd-8793-37ede30ff32b.png)
![output green](https://user-images.githubusercontent.com/33333343/170553982-611c7867-1836-45ad-9af3-0d6bb482e0dc.png)
![output brown](https://user-images.githubusercontent.com/33333343/170553984-36af4455-163a-4828-9743-29295d8cc6d3.png)
![output blue](https://user-images.githubusercontent.com/33333343/170553986-18f19341-1424-4a09-9fb7-257b3710f1f5.png)
![background](https://user-images.githubusercontent.com/33333343/170553987-ec832f02-eef4-413f-9e81-a105d3d39b79.jpg)
![output output red](https://user-images.githubusercontent.com/33333343/170554008-b7f34d4b-7499-44e9-b181-daf2305f210d.png)
![output output output blue](https://user-images.githubusercontent.com/33333343/170554019-db5c0549-2e00-42c2-bcf6-9153c79ae6cd.png)
![output output orange](https://user-images.githubusercontent.com/33333343/170554026-1cf0bb40-2298-4789-9af5-129b338cad4a.png)
![output output green](https://user-images.githubusercontent.com/33333343/170554030-6ace7383-1718-405b-9053-1bb7958ba9da.png)
![output output brown](https://user-images.githubusercontent.com/33333343/170554035-b1135ffe-1b57-4a38-89dd-f43b1f40416f.png)
![output output blue](https://user-images.githubusercontent.com/33333343/170554040-18818f54-7642-4908-98db-b3ca73547d94.png)
![output red](https://user-images.githubusercontent.com/33333343/170554045-59ae15b6-db82-45f4-b4d8-836e1803bc4f.png)

**Comment by snibgo** [comment]
Your code is:
```
FOR %G IN ("*.png") DO magick "%G" background.gif ( *.png -resize 800x990 ) -gravity Center -geometry +30-160 -composite "output.%G"
```
So your magick command runs once for each png file. Each time it runs, it reads *all* the png files, "\*.png". Is that what you want to do? Perhaps you want to re-read just "%G".

**Reply by soyebravat** [reply]
Yes precisely. I want to use the background.gif command as a base background and have all the pngs run once over the background with the parameters set. I think my code could be wrong based on the %G. Any ideas how I can make it work?

**Comment by snibgo** [comment]
I suggest you start by writing the magick command you need to combine one foreground with one background. I don't know what result you want, so I don't know what that command should be. But it should read only two images, and will write one image. For example:
```
magick "bedroom.jpg" "yellowBirds.png" -composite "out.jpg"
```
When that command does what you want, you can put that in a "for" loop to process all the different foreground images. For example:
```
for %G in (*.png) do magick "bedroom.jpg" "%G" -composite "output_%G"
```

**Reply by soyebravat** [reply]
So this code works perfectly: `magick "background.jpg" "output.red.png" ( *.png -resize 800x990 ) -gravity north -geometry +30-160 -composite "result2.jpg"`

Problem is how do i get the batch to work and process all files or can i possibly use mogrify?

**Comment by fmw42** [comment]
That command does not look correct to me. It has too many images.  What is out.red.png?  The third image in a composite command should be the mask. If you want to composite a bunch of images with a background. then you will need to put null: between them and use -layer composite rather than composite.

magick background null: ( *.png ... ) -compose XXX -layers composite result_%d.jpg

This will make one output for every png image used.  Is that what you are trying to do?  If not, then explain further.

**Reply by soyebravat** [reply]
Hi fmw42, 

That was a massive help!!!

Thank you so much, exactly what I wanted. works perfectly!
